NCMP_for_2-1-1_and_2-2-1/WHO_code_test.R

```diff
@@ -0,0 +1,97 @@
+clean_NCMP <- NCMP_data[NCMP_data$Height > 0, ]
+
+# remove rows with suppressed imd and perform check that imd is in expected range
+clean_NCMP <- clean_NCMP[clean_NCMP$suppress_imd != "Y", ]
+if (any(unique(clean_NCMP$schoolindexofmultipledepriv) %in% c(1:10)) == FALSE){
+    print("ERROR! Check that schoolindexofmultipledepriv variable ranges from 1 to 10")
+  }
+
+#check how many entries were lost after all the cleaning:
+# This is just for testing purposes, raw data should need no cleaning
+lost_data <- paste("Cases:", nrow(clean_NCMP)-nrow(NCMP_data), ",", 
+                   "Percent:", (nrow(NCMP_data)-nrow(clean_NCMP)) / nrow(NCMP_data) * 100) # as above, in %
+lost_data
+
+# regroup imd variable from deciles into quintiles to fit the WHO function requirement
+clean_NCMP$school_imd_quintile <- ceiling(clean_NCMP$schoolindexofmultipledepriv / 2)
+
+if (any(unique(clean_NCMP$school_imd_quintile) %in% c(1:5)) == FALSE){
+  print("ERROR! Check that school_imd_quintile variable ranges from 1 to 5")
+}
+
+
+#main function, takes around 26 mins to run:
+
+#the wealthq argument is fed the regrouped IMD variable and assumes 1 = most deprived, 5 = least deprived
+#typeres argument expects urban/rural breakdown variable, 
+# which is not available in the test data but is reported to be available in the raw data as "PupilUrbanRuralIndicator"
+# schoolgovernmentofficeregion variable is used here as a test for othergr argument (which may take ethnicity from the raw data)
+# reformatting this variable into character, in case it's a factor (to avoid function errors)
+clean_NCMP$schoolgovernmentofficeregion <- as.character(clean_NCMP$schoolgovernmentofficeregion)
+
+test_NCMP <- anthro_prevalence(sex = clean_NCMP$GenderCode,
+                             age = clean_NCMP$AgeInMonths, 
+                             is_age_in_month = TRUE, 
+                             weight = clean_NCMP$Weight, 
+                             lenhei = clean_NCMP$Height, 
+                             measure = "H",
+                             wealthq = clean_NCMP$school_imd_quintile,
+                             othergr = clean_NCMP$schoolgovernmentofficeregion)
+
+#get relevant info only
+rownames(test_NCMP) <- test_NCMP$Group #assigning the first column as rownames (makes is cleaner)
+test_NCMP <- test_NCMP[, -1] #removing the duplicate column
+
+#lots of NA rows for the non-used function arguments, so getting rid of those:
+test_NCMP <- test_NCMP[rowSums(is.na(test_NCMP)) != ncol(test_NCMP), ]
+
+#renaming the default deprivation variable to match dataset:
+first_Q <- which(rownames(test_NCMP)=="Wealth quintile: Q1: Poorest")
+rownames(test_NCMP)[first_Q:(first_Q + 4)] <- c("School IMD Q1: most deprived", 
+                                                "School IMD Q2",
+                                                "School IMD Q3","School IMD Q4",
+                                                "School IMD Q5: least deprived")
+
+
```

Comment:
should we do something similar to rename Other grouping to something more meaningful?

Reply:
Yes, that would be good, but I'm not sure I can do this in advance. The `othergr `argument for the final script will be applied on the ethnicity variable "Ethnicity_desc" (not available in the test data set) and I don't know exactly how the different ethnicities are labelled in the raw data. All I know is that it represents 17 ethnic groups, with text descriptions e.g.  White - British. So there will be 17 additional rows, instead of the `schoolgovernmentofficeregion` used here...

Reply:
@ANikolova22 Oh ok, didn't realise that. So does that mean we wont be able to report region as well? Can we have both?

Reply:
So, yes - maybe... we will get urban/rural breakdown (again, not included in the test code, but it is in the final code). In the raw data there is a variable [PupilRegionCode_ONS], which is "ONS region code based of child postcode". We could potentially feed that into the `gregion` argument of anthro_prevalence in addition to the other variables. 

I'm just worried about starting to become disclosive if we request too much granularity (though I guess we can try and see what they say). I don't know how many levels that variable has - if it is like the `schoolgovernmentofficeregion ` in the test code, then won't be too many, but if it is like the `schooltier2localauthoritycode`, that would be 326 rows more in the output csv

Reply:
OK cool :) 

PupilRegionCode_ONS sounds to me like it will be region (e.g. Yorkshire and the Humber, East of England etc), so should be fine. I guess the problem with too high a granularity will only occur in the cross disaggregations (e.g. sex by ethnicity by region), so we could say we are aware that this may be an issue. If it is, please can they drop the cross-disaggregations that are problematic, but keep the simple disaggregations ie. drop sex by ethnicity by region, but keep sex, region, ethinicity, sex by region, sex by ethnicity, and (if possible) region by ethnicity. How does that sound?

Reply:
Yes, sounds good (will add `gregion = PupilRegionCode_ONS` to the main script)! On second thought... it shouldn't be a problem because the function doesn't actually do detailed cross-disaggregations by default (sadly), other than sex*age. The test output for IMD and region give the overall prevalence numbers